src/sentry/integrations/jira_server/client.py

```diff
@@ -125,10 +140,49 @@ def request(self, *args, **kwargs):
                 signature_type='auth_header')
         return self._request(*args, **kwargs)
 
+    def get_cached(self, full_url):
+        """
+        Basic Caching mechanism for requests and responses. It only caches responses
+        based on URL
+        """
+        # TODO(mark) This key needs to come from the 'jira style'
+        key = 'sentry-jira-server:' + md5(full_url, self.base_url).hexdigest()
+        cached_result = cache.get(key)
+        if not cached_result:
+            cached_result = self.get(full_url)
+            cache.set(key, cached_result, 60)
+        return cached_result
+
     def get_valid_statuses(self):
         # TODO Implement this.
         return []
 
     def get_projects_list(self):
         # TODO Implement this
         return []
+
+    def get_create_meta(self, project=None):
+        params = {'expand': 'projects.issuetypes.fields'}
+        if project is not None:
+            params['projectIds'] = project
+        return self.get(
+            self.META_URL,
+            params=params,
+        )
+
+    def get_priorities(self):
+        return self.get_cached(self.PRIORITIES_URL)
+
+    def get_versions(self, project):
+        return self.get_cached(self.VERSIONS_URL % project)
+
+    def search_issues(self, query):
+        # check if it looks like an issue id
+        if re.search(r'^[A-Za-z]+-\d+$', query):
```

Comment:
So, I wondered about this. The issue id is the project key and a number, to my knowledge. I played around with jira and realized that the project key can have numbers. The first must be an uppercase letter then the next ones can be alphanumeric (upper case letters only). Lower case letters are not allowed.

Reply:
I'd like this method to go away when jira clients are combined. I can update the regex then.